Samples/BlazorExample/BlazorExample/Client/Pages/EditPerson.razor

```diff
@@ -62,10 +62,20 @@ else
   [Parameter]
   public string id { get; set; }
 
-  protected override async Task OnInitializedAsync()
+  protected override void OnInitialized()
   {
     vm.Saved += () => NavigationManager.NavigateTo("listpersons");
+     vm.ModelChanging += (o, n) =>
+    {
+      if (o != null)
+        o.PropertyChanged -= async (s, e) => await InvokeAsync(() => StateHasChanged());
+      if (n != null)
+        n.PropertyChanged += async (s, e) => await InvokeAsync(() => StateHasChanged());
```

Comment:
I've been wondering about the efficiency of calling StateHasChanged at this level. This is marking the top-level component as needing re-rendering on every change to any property. I think for most small to medium edit forms that would be fine, but on a very large edit form with lots of properties, this would make the re-render and diffing work harder than necessary.

I wonder if an alternative is to push the choice to re-render to a lower-level component, and only calling StateHasChanged within that child if the property with the which the component is associated has changed, as I believe we are passed the property name in the EventArgs of PropertyChanged.

The performance difference might be very small, and this might therefore be unnecessary. However, I seem to remember hearing of people having problems with React and/or Angular being too slow when an update is made on a very complex component graph. That was a JS performance issue, but at the moment Mono isn't any faster than JS; indeed, it appears it's about half the speed of JS on average because Mono is currently operating in interpreted mode when running on wasm.